crates/hulk_replayer/src/user_interface.rs

```diff
@@ -17,11 +23,32 @@ pub struct ReplayerApplication {
     timing: Timing,
     viewport: Viewport,
     position: f32,
+
+    rerun_request: Receiver<()>,
 }
 
 impl ReplayerApplication {
-    pub fn new(replayer: Replayer<ReplayerHardwareInterface>) -> Self {
+    pub fn new(context: &CreationContext, replayer: Replayer<ReplayerHardwareInterface>) -> Self {
         let timing = join_timing(&replayer);
+        let (rerun_request_sender, rerun_request) = channel();
+        let context = context.egui_ctx.clone();
+        let parameters_updated = replayer.get_parameters_changed();
+
+        spawn(move || {
+            let runtime = tokio::runtime::Builder::new_current_thread()
+                .build()
+                .unwrap();
+
+            runtime.block_on(async move {
+                loop {
+                    parameters_updated.notified().await;
+                    println!("requesting re-replaying");
+                    rerun_request_sender.send(()).expect("channel closed");
+                    context.request_repaint();
+                }
+            });
+        });
```

Comment:
Or move the replaying into this thread. The fact that it currently is evaluated synchronously is also smelly. The semantic of replaying is fire-and-forget from the viewpoint of the UI, so that would make sense.

Reply:
I did this and much more, see updated description.

Reply:
Thanks